Mathlib/RingTheory/UniqueFactorizationDomain.lean

```diff
@@ -2150,4 +2150,13 @@ lemma _root_.induction_on_primes {P : ℕ → Prop} (h₀ : P 0) (h₁ : P 1)
   · exact fun a p _ hp ↦ h p a hp.nat_prime
 #align induction_on_primes induction_on_primes
 
+lemma prime_composite_induction {P : ℕ → Prop} (zero : P 0) (one : P 1)
```

Comment:
This looks like it belongs with `recOnPrimeCoprime`, and probably proved using things there.